composer/datasets/coco.py

```diff
@@ -0,0 +1,597 @@
+                c.chunk(nm)))
+    if isinstance(x, List) and isinstance(y, List) and isinstance(a, List) and isinstance(b, List) and isinstance(
+            c, List):
+        return list(
+            zip(
+                [x[i::n_microbatches] for i in range(n_microbatches)],
+                [y[i::n_microbatches] for i in range(n_microbatches)],
+                [a[i::nm] for i in range(nm)],
+                [b[i::nm] for i in range(nm)],
+                [c[i::nm] for i in range(nm)],
+            ))
+
+
+def _isArrayLike(obj):
+    return hasattr(obj, '__iter__') and hasattr(obj, '__len__')
+
+
+class COCODetection(data.Dataset):
+
+    def __init__(self, img_folder, annotate_file, transform=None):
+        self.img_folder = img_folder
+        self.annotate_file = annotate_file
+
+        # Start processing annotation
+        with open(annotate_file) as fin:
+            self.data = json.load(fin)
+
+        self.images = {}
+
+        self.label_map = {}
+        self.label_info = {}
+        start_time = time.time()
+        # 0 stands for the background
+        cnt = 0
+        self.label_info[cnt] = "background"
+        for cat in self.data["categories"]:
+            cnt += 1
+            self.label_map[cat["id"]] = cnt
+            self.label_info[cnt] = cat["name"]
+
+        # build inference for images
+        for img in self.data["images"]:
+            img_id = img["id"]
+            img_name = img["file_name"]
+            img_size = (img["height"], img["width"])
+            if img_id in self.images:
+                raise Exception("dulpicated image record")
+            self.images[img_id] = (img_name, img_size, [])
+
+        # read bboxes
+        for bboxes in self.data["annotations"]:
+            img_id = bboxes["image_id"]
+            category_id = bboxes["category_id"]
+            bbox = bboxes["bbox"]
+            bbox_label = self.label_map[bboxes["category_id"]]
+            self.images[img_id][2].append((bbox, bbox_label))
+
+        for k, v in list(self.images.items()):
+            if len(v[2]) == 0:
+                # print("empty image: {}".format(k))
+                self.images.pop(k)
+
+        self.img_keys = list(self.images.keys())
+        self.transform = transform
+
+    #@property
+    def labelnum(self):
+        return len(self.label_info)
+
+    @staticmethod
+    def load(pklfile):
+        with bz2.open(pklfile, "rb") as fin:
+            ret = pickle.load(fin)
+        return ret
+
+    def save(self, pklfile):
+        with bz2.open(pklfile, "wb") as fout:
+            pickle.dump(self, fout)
+
+    def __len__(self):
+        return len(self.images)
+
+    def __getitem__(self, idx):
+        img_id = self.img_keys[idx]
+        img_data = self.images[img_id]
+        fn = img_data[0]
+        img_path = os.path.join(self.img_folder, fn)
+
+        from os.path import exists
+
+        if (exists(img_path)):
```

Comment:
This if-statement seems sus... Are there image paths that don't exist? If that is the case, I think there is a better way to handle this